### ISSUE TITLE
Design Intent - Accessibility Feedback - Identity Core Team, Terms of Use (Terms and Conditions)

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#61712]
## VFS actions
- [ ] Assign this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] Comment on this ticket:
  - [ ] If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - [ ] When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - [ ] When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] Close the ticket when all feedback has been addressed.

## Thoughts/questions

- Sorry for missing the meeting on Friday! Please feel free to attend Collab Cycle office hours or to schedule me for a meeting if you want to discuss any of this.

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- **Must:** After the scrollable block of legal text, you'll need to have another heading for the section of content with the checkboxes and agree/cancel buttons. Screen reader users often navigate the page by jumping straight to a heading they care about. If a screen reader user doesn't care about the legal text and just wants to get to the actions to get past this screen, the closest they can get would be to jump to the last heading of the scrollable block, then get through the last section of content (however long that might be) before getting to the checkboxes. Giving them a heading before the checkboxes makes navigating within the page a lot easier.
- **Must:** Many screen reader users may assume that the agree/cancel buttons are the end of the page, and won't have the visual cue that there's more beneath those buttons that they may want to read. Since the text that comes after those buttons may impact a user's decision to agree or not, they need to be aware of it before they reach those buttons. For trying to keep the buttons above the fold, maybe an additional info component ("What happens if I don't agree?") before the checkboxes might work here.
- **Must/It Depends:** In the meeting Sara brought up whether the cancel button should be a link or a button. The goal here is material honesty about what the element is doing, since some assistive technologies handle links and buttons differently. If its purpose is navigational, it should be a link. If it takes some action, it should be a button. In this case, I'm not sure! Here are the cases as I understand them:
   - If the only consequence of clicking Cancel is to log you out and navigate you to the "We've signed you out" page, then it should be a link. This is consistent with the log out link in the user menu when you're authenticated.
   - If clicking Cancel also does something, like saving information to your profile (some flag that says "this user said no to our terms and conditions"), you're saving user data and that's an action so it should be a button.
- **Must:** In the meeting the idea of making the agree/cancel buttons disabled came up. Disabling elements like that is a bad pattern for screen reader users, so I'll say definitely don't do that! If there's something that's required in order to proceed with those actions (check the boxes, scroll all the way to the bottom of the box, anything else) the best way to handle that is with an error message that appears when a user attempts to agree. That way the buttons are always discoverable and users get feedback on why they can't proceed, rather than having to guess about what it takes to make the button work.
- **Should:** I have concerns about how the scrollable block of legal text will work. In the meeting Sara and Martha both brought up mobile, and I'll bring up zoom and screen magnification for low-vision users. Per WCAG guidance, we test everything at up to 400% browser zoom. At high zoom you don't have a ton of screen real estate to work with, which you may not want to take up with a scrollbar and the padding around/inside the box. You may also run into issues if the scrollable box is larger than the viewport
   
   One possibility would be to only have the box be fixed-height and scrollable on sufficiently large viewports, then have it be full-size and not scrollable on everything else? I don't know if that's desirable. But keeping the action buttons above-the-fold becomes very challenging when you start zooming in, so if stakeholders are okay dropping that expectation for smaller devices/high zoom that might be an option.
   
   But note that users of screen magnification devices or software will not trigger the CSS breakpoints that you get when you use your browser zoom. In those cases, it's possible that a user at high magnification may reach the end of the visible text of the box while the scrollbars are outside of their view. This puts some cognitive load on the user to recognize what the box is and how it's supposed to behave. It's not a "must," but I would be interested in whether there's another way to implement this using collapsible content (additional info or accordions) or something like that.
- **Should/Engineering Note:** As currently coded, I don't believe the [`va-checkbox` component](https://design.va.gov/storybook/?path=/docs/components-va-checkbox--default) supports including a hyperlink in the label text. And while it's not explicitly disallowed by WCAG or the VA.gov experience standards, I'd like to discourage you from doing that too.
   
   HTML checkboxes can be checked or un-checked either by clicking the box itself or by clicking the label associated with the box. For users with limited fine motor control, this serves to make the target size of the box bigger and make it easier for them to successfully check the box. When a link is included in the label text, a click/touch action on the link activates the link instead of checking the box. Now you've got two targets inline with each other and with no margin between them, which might lead to a user accidentally checking the box when they mean to follow the link, or vice versa. With limited fine motor control, correcting errors isn't always easy to do.
   
   With the length of the label text and the portion of the text that's linked, I don't think this is a super high likelihood issue. But it's a possible issue in a workflow that every VA.gov authenticated user will have to go through, so my "should" recommendation here would be to separate the link out from the checkbox. (And if they're separated, the link should come before the checkbox so users are prompted to read before they decide whether to check the box.)

**Comment by asg5704** [comment]
@briandeconinck I'm going to summarize based on your recommendations
- [ ] Create a Heading before the "Accept/Cancel" portion to allow screen reader users the ability to skip the ToU content
> Any suggestions as to what this heading should be?
- [ ] Add an "info component aka va-alert info" to state "What happens if they don't accept" before the "Accept/Decline" actions. Hierarhical changes so that users will understand the actions better before taking them.
> Should this be an Info or a Warning?
- [ ] The "Cancel" will be a button. The "Cancel" button will make an API request to the user's preference about not accepting to the Terms of Use before logging them out and finally redirecting them to the "We signed you out because..." page
> I would prefer this be a button instead of a link (because of what is stated above). However, if it is determined that this should be a link, then the "Sign in buttons" within the Sign in Modal should also become links AND design.va.gov should also be updated to take a strong stance on it as well so we are consistent across the design ecosystem.
- [ ] Do NOT disable buttons and instead create "error content" if a user is required to check a box (secondary action) before clicking "Agree/Cancel" (primary action)
> I agree here (also see the note below about checkboxes)
- [ ] ToU content
  - [ ] In mobile, there are two options: 1.) Allow ToU content to flow freely, or 2.) Wrap ToU content in an Accordion or Collapsible content
  - [ ] In desktop, there are three options: 1.) Fixed height scrollable content (in mock up), 2.) Content flow freely, or 3.) Wrap ToU content in an Accordion or Collapsible content
> I'm also wondering the best course of action here. While the full content is completely accessible it also unsightly and takes up a lot of screen real estate. Would it make sense to have each ToU section as an accordion that way we can chunk the different sections appropriately? Looking for guidance.
- [ ] The checkboxes should NOT have links in the label due to cognitive overload and mishandling of actions by users. Instead provide any links before the check box action.

Note: The checkboxes as it stands right now are completely optional and could be removed if we have sufficient data to support that there is no need for them, looking for guidance on whether they should be removed or not.


This is what I have based on your feedback, take note that I removed the checkboxes pending your recommendations on them. Thank you!
![Based on a11y](https://github.com/department-of-veterans-affairs/va.gov-team/assets/67602137/b7c25531-ac4e-4bd9-9cdc-cc02b1ec567a)
